addons/stock/report/stock_forecasted.py

```diff
@@ -307,13 +309,26 @@ def _reconcile_out_with_ins(lines, out, ins, demand, product_rounding, only_matc
         for out in outs:
             outs_per_product[out.product_id.id].append(out)
 
-        ins_per_product = defaultdict(list)
+        ins_per_product = defaultdict(OrderedSet)
         for in_ in ins:
-            ins_per_product[in_.product_id.id].append({
-                'qty': in_.product_qty,
+            in_data = {'qty': in_.product_qty,
                 'move': in_,
-                'move_dests': in_._rollup_move_dests()
-            })
+                'move_dests': in_._rollup_move_dests(),
+                'priority': -int(in_.priority),
+                'date': in_.date,
+                'id': in_.id,
+                }
+            map_in_id_to_data[in_.id] = {'data': in_data}
```

Comment:
Do you really need a 'data' subdict ? Can't you just map `[in_.id] -> in_data]` and do a `dict.update` or something similar ?